apps/src/code-studio/pd/application/facilitator1819/Facilitator1819FormComponent.jsx

```diff
@@ -19,7 +24,15 @@ export default class Facilitator1819FormComponent extends FormComponent {
       console.warn(`Label missing for ${name}`);
       return name;
     }
-    return this.constructor.labels[name];
+
+    const label = this.constructor.labels[name];
```

Comment:
I'm not entirely sure I understand how this `labels` stuff works. Storing things on `this.constructor` feels strange to me, but I'm don't know that I really understand what's happening.

Reply:
It's just accessing the [static field](https://github.com/code-dot-org/code-dot-org/blob/staging/apps/src/code-studio/pd/application/facilitator1819/Section1QuestionsAboutYou.jsx#L18). I tried `Facilitator1819FormComponent.labels` instead, but then it didn't return the inherited value. I suppose this could back up the argument that we should avoid inheritance in React ;) 

Is there a preferred way to access a static in js? Or I suppose I could set it to an instance variable in the constructor if that's cleaner?

Reply:
MDN [recommends](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Classes/static) `this.constructor`

> Static methods are not directly accessible using the this keyword from non-static methods. You need to call them using the class name: CLASSNAME.STATIC_METHOD_NAME() or by calling the method as a property of the constructor: this.constructor.STATIC_METHOD_NAME().

Reply:
In general, something like `Facilitator1819FormComponent.labels` is the way that I would expect to use statics.

Would it make sense to just have these labels as props instead of statics?

Reply:
Yeah, but then like I said it will always return `{}`. I want to get the derived version. Props don't make sense in this case because it's not passed in from the caller. I want these to just be associated with the control itself, for example [here](https://github.com/code-dot-org/code-dot-org/blob/staging/apps/src/code-studio/pd/application/facilitator1819/Section1QuestionsAboutYou.jsx#L18). 

I could try rewriting this with composition (still using the base `FormComponent` but not this intermediate `Facilitator1819FormComponent` class) and pass props from the wrapper.

Reply:
Okay, I think I better understand how this is all happening now. I think a lot of my struggles from reading this come from the use of inheritance instead of composition, which leads to set of components that are much more intertwined than most React components (i.e. it becomes much more difficult to understand a single component in isolation).

I'll leave it to you whether it's worth rewriting. My inclination might be to explore doing so, though I'm admittedly missing a lot of context here, so my instincts might be completely wrong. My experience with inheritance in JS in general is that (a) it ends up being confusing (b) it becomes especially confusing when you start introducing multiple levels of inheritance, which I believe has happened here with `Section1QuestionsAboutYou` deriving from `Facilitator1819FormComponent` deriving from `FormComponent`

Reply:
I'm going to merge this as-is, and if take a look at refactoring with composition later if I have some time, since it's not entirely trivial at this point.